modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/NodeImpl.java

```diff
@@ -2243,10 +2243,6 @@ public Message handleAppendEntriesRequest(final AppendEntriesRequest request, fi
                         .success(false)
                         .term(this.currTerm);
 
-                if (request.timestamp() != null) {
```

Comment:
If I'm not mistaken, transaction processing will cause other nodes' clocks to be adjusted to the sender clock (due to clock updates that happen when we process replica messages), and that's all.

1. Are there any other sources of such adjustments?
2. Will we adjust clocks if there is no transactional traffic? Will partition safe time propagation cause such adjustments?
3. If there are no partitions in the cluster at all, will the clocks be adjusted? Should we leave the 'through Raft adjustment' for the Metastorage?

Reply:
It would be nice to have detailed explanation why it's safe. It took some time for me to prove that it's true because of `tx1.commitTs < A.lease.endTime < B.lease.startTime < tx2.commitTs`.